File: annotationprocessortesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/Week.java
package com.tickaroo.tikxml.annotationprocessing.textcontent;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import java.util.List;
import java.util.Objects;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Week {

  @Element
  List<Day> days;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof Week)) return false;

    Week week = (Week) o;

    return Objects.equals(days, week.days);
  }

  @Override public int hashCode() {
    return days != null ? days.hashCode() : 0;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/Week.kt
package com.tickaroo.tikxml.annotationprocessing.textcontent

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
class Week {

    @Element
    var days: List<Day>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Week) return false

        val week = other as Week?

        return if (days != null) days == week!!.days else week!!.days == null
    }

    override fun hashCode(): Int {
        return if (days != null) days!!.hashCode() else 0
    }
}
